relayer_api/services/relayerService.js

```diff
@@ -118,8 +117,9 @@ class RelayerService {
 		let reward = utils.bigNumberify(_reward);
 		let wei = utils.bigNumberify(_wei);
 
-		const counterfactualData = transactionsStorage.getData(identityAddress);
-		if (counterfactualData) {
+		let record = await databaseManager.findRecord(identityAddress)
+
+		if (record) {
```

Comment:
Should we check if someone has already executed this counterfactual transaction?